benchmarks/work_precision_sets/time_vs_abstols.py

```diff
@@ -0,0 +1,126 @@
+import pybamm
+import numpy as np
+import matplotlib.pyplot as plt
+
+
+parameters = [
+    "Marquis2019",
+    "Prada2013",
+    "Ramadass2004",
+    "Chen2020",
+]
+models = ["SPM", "DFN"]
+abstols = [
+    0.001,
+    0.0001,
+    1.0e-5,
+    1.0e-6,
+    1.0e-7,
+    1.0e-8,
+    1.0e-9,
+    1.0e-10,
+    1.0e-11,
+    1.0e-12,
+    1.0e-13,
+]
+solvers = ["IDAKLU", "Casadi - safe", "Casadi - fast"]
+
```

Comment:
You can define the figure outside the `for` loop and then loop over the axes (see https://stackoverflow.com/questions/20288842/matplotlib-iterate-subplot-axis-array-through-single-list).

You can also get all the possible combinations of solvers and models by using `itertools.product` (see https://stackoverflow.com/questions/533905/get-the-cartesian-product-of-a-series-of-lists).

If you zip the `axes` and the output of `itertools.product`, you can get rid of lines 68-85. Hopefully, this way it will also be more generalisable (i.e. if we want to add a model).